File: musicShopProject/Model/Manufacturers/ManufacturerBlank.cs
using musicShopProject.Model.Manufacturers.Enums;

namespace musicShopProject.Model.Manufacturers;

public partial class ManufacturerBlank
{
    public Guid? Id { get; set; }

    public String? Name { get; set; }

    public String? Logo { get; set; }

    public Countries? Country { get; set; }
}

public partial class ManufacturerBlank
{
    public class Validated
    {
        public Guid Id { get; }

        public String Name { get; }

        public String Logo { get; }

        public Countries Country { get; }

        public Validated(Guid id, String name, String logo, Countries country)
        {
            Id = id;
            Name = name;
            Logo = logo;
            Country = country;
        }
    }
}

File: musicShopProject/ClientApp/src/domain/manufactures/manufacturerBlank.ts
import { Countries } from "./countries";
import { Manufacturer } from "./manufactured";

export interface ManufacturerBlank{
    id: string | null,
    name: string | null,
    logo: string | null,
    country: Countries | null
}

export namespace ManufacturerBlank {
    export function getEmpty(): ManufacturerBlank {
        return {
            id: null,
            name: null,
            logo: null,
            country: null
        };
    }

    export function toProductBlank(category: Manufacturer): ManufacturerBlank {
        return {
            id: category.id,
            name: category.name,
            logo: category.logo,
            country: category.country
        };
    }
 }